components/merkledb/src/access/extensions.rs

```diff
@@ -46,7 +46,9 @@ pub trait AccessExt: Access {
         I: FromAccess<Self>,
     {
         // We know that `Group` implementation of `Restore` never fails
-        Group::from_access(self, IndexAddress::with_root(name)).unwrap()
+        Group::from_access(self, IndexAddress::with_root(name)).unwrap_or_else(|e| {
```

Comment:
I believe that since we don't use `format!(..)`, `expect` is sufficient here, since it contains the error message ([1](https://doc.rust-lang.org/src/core/result.rs.html#957-962), [2](https://doc.rust-lang.org/src/core/result.rs.html#1164)) as well.

Reply:
`expect` would trigger clippy's [expect_fun_call](https://rust-lang.github.io/rust-clippy/master/#expect_fun_call) lint.

Reply:
Why? It gets triggered if `format` is used, and here it will be just `.expect("MerkleDB error")` (as I mentioned, error message will be added automatically.

Reply:
You are right, changed to expect.

Reply:
`expect` uses a `Debug` representation of an error though, while we might want a `Display` one.

Reply:
Do we? I thought that for `panic` (which is not meant to be seen by end-side users) should contain `Debug` representation. It's sort of similar to the python `__repr__`/`__str__`: the first is meant to be seen by a programmer (and should be unambiguous), and the latter is meant for users.

Reply:
I consider service developers to belong to the "users" category. This is debatable, but I'd argue that a service developer seeing an error like

```text
Error accessing `currency.wallets`: Wrong index type: expected ProofMapIndex, but got MapIndex
```

is more likely to understand / fix it than if she sees 

```text
AccessError { addr: IndexAddress { name: "currency.wallets", bytes: None }, kind: WrongIndexType { expected: ProofMapIndex, actual: MapIndex } }
```

...especially if we're talking about Java developers.

Reply:
Well, makes sense. I'm not against it.

Reply:
👌